### PR TITLE
Update manual partition reassignment steps

### PR DESCRIPTION
**Pull request by daisukebe**
Since v23.3, RPK has ability to move partition replicas across brokers, and also cores. We don't need to use the third party tool any more. This commit updates the document accordingly.

Related, https://github.com/redpanda-data/redpanda/issues/1227

Preview: 
https://deploy-preview-160--redpanda-docs-preview.netlify.app/23.3/manage/cluster-maintenance/cluster-balancing/

### CODE REVIEW

**Comment by micheleRP** [comment]
Thank you, @daisukebe! I'll run through a doc edit now. But if this is for 23.3, it should go in `v-WIP/23.3` instead of `main`.

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]:
```suggestion
. For example, to change the replica set of partition 1 from `[0 1 2]` to `[3 1 2]`, and to change the replica set of partition 2 from `[0 1 3]` to `[2 1 3]`, run:
```

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]:
```suggestion
To cancel all in-progress partition reassignments, run `move-cancel`:
```

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]:
```suggestion
To cancel specific movements to or from a given node, run:
```

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]: @Daisuke, should we now remove this section (_Differences in partition balancing between Redpanda and Kafka_)? If this is is still useful to customers, then please move it up a level, from `===` to `==`. 
Please also move the following section (_Assign partitions at topic creation_) up a level too, from `===` to `==`.

**Comment by Deflaimun** [comment]
Changed base to v-WIP/23.3 since the content belongs to beta/new version

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]:
```suggestion
As an alternative to Redpanda partition balancing, you can change partition assignments explicitly with `rpk cluster partitions move`.
```

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]:
```suggestion
To reassign partitions with `rpk`:
```

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]:
```suggestion
. Set the `partition_autobalancing_mode` property to `off`. If Redpanda partition balancing is enabled, Redpanda may change partition assignments regardless of what you do with `rpk`.
```

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/cluster-balancing.adoc:118]:
```suggestion
Check the movement status with `rpk cluster partitions move-status`, and see new assignments with `rpk topic describe -p TOPIC`.
```

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]:
```suggestion
. Verify that the reassignment is complete with `move-status`:
```

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]:
```suggestion
. Change partition assignments. For example, to change the replica set of partition 1 from `[0 1 2]` to `[3 1 2]`, and to change the replica set of partition 2 from `[0 1 3]` to `[2 1 3]`, run:
```

**Comment by daisukebe** [comment]
> Changed base to v-WIP/23.3 since the content belongs to beta/new version

Great, thank you @Deflaimun!

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (daisukebe) [reply modules/manage/pages/cluster-maintenance/cluster-balancing.adoc:118]: Ah @micheleRP, this is an command output, thus it's not a good idea to change the wording.

**Comment by daisukebe** [comment]
@micheleRP most of all suggestions look good to me, except for one I mentioned above. Much appreciated!

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (daisukebe) [reply modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]: Some still look good but some look obsolete. Let me chase.

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]:
```suggestion
== Manually move partitions
```
@daisukebe what do you think about changing this heading to _Manually move partitions_ to align it with the heading earlier on this page _Manually change leadership_?

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (micheleRP) [reply modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]: @Deflaimun the preview for this PR still is showing in main, not Beta. Is it OK?

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (daisukebe) [reply modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]: Totally makes sense!

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (daisukebe) [reply modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]: @micheleRP , I've updated the differences in https://github.com/redpanda-data/docs/pull/160/commits/16ad33381bc2f7cd217821319be046884d8b3a0f. Who was the original SME for this list? I'd like to have the person involved for review.

Comment (micheleRP) [reply modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]: Nyalia worked on that exact section, and I believe John Spray later contributed. SMEs for other content on this page = @ztlpn and @mmaslankaprv

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]:
```suggestion
* Kafka doesn't support shard-level partition assignments, but Redpanda does. For example, to specify that all replicas of partition `1` be assigned on shard `0` on each broker, and all replicas of partition `2` be assigned on shard `1` on each broker, run: `rpk cluster partitions move test -p 1:3-0,1-0,2-0 -p 2:2-1,1-1,3-1`. For more information, run: `rpk cluster partitions move --help`.
```

**Comment by Feediver1** [comment]
Couple of items here:
- @mattschumpert Docs never had a ticket requesting this update. So we are trying to get this done (Thanks @daisukebe !) before the release.
- We absolutely need a thorough PM review/approval of this material. I've asked @micheleRP not to merge this until PM has reviewed and approved.
Thanks!

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (mattschumpert) [inline modules/manage/pages/cluster-maintenance/cluster-balancing.adoc:125]: We probably want to put this alternative syntax before the statement about checking the move-status (above)

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (mattschumpert) [inline modules/manage/pages/cluster-maintenance/cluster-balancing.adoc:128]: actually, this whole section like a duplicate to the statement above and a better version of it.  perhaps we can just remove the statement above .

> Check the movement status with `rpk cluster partitions move-status`, and see new assignments with `rpk topic describe -p TOPIC`.

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (mattschumpert) [inline modules/manage/pages/cluster-maintenance/cluster-balancing.adoc:87]: @micheleRP Im not sure we should completely eliminate the Kafka API bit here, because some customers have asked for that by name, and if you're using Kafka ecosystem tools to automate this they're using the API nor RPK. 

 We may want to just leave a note at the bottom says "If you prefer, Redpanda also supports the use of the AlterPartitionAssignments Kafka API and using standard kafka tools such as  kafka-reassign-partitions.sh.

Or we may just want to keep this whole API section and put it at the bottom as its kind of the last resort option (or if your Kafka tooling depends on it)

**Review by mattschumpert — APPROVED** [review]

**Comment by mattschumpert** [comment]
@Feediver1 left a couple of comments and approved.  Between these comments and Daisuke's review that should be good.

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (daisukebe) [reply modules/manage/pages/cluster-maintenance/cluster-balancing.adoc:125]: Sorry for the confusion, but `the statement about checking the move-status` is an output of the `move` command for guiding users appropriately. Thus it is documented as expected.

https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/cli/cluster/partitions/move.go#L204

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (ztlpn) [inline modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]: In 24.1 we are going to get rid of manual shard assignments and let redpanda choose the shard automatically in all cases. So I wouldn't go into too much detail about this.

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (ztlpn) [reply modules/manage/pages/cluster-maintenance/cluster-balancing.adoc:87]: agreed

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (daisukebe) [reply modules/manage/pages/cluster-maintenance/cluster-balancing.adoc:128]: > Check the movement status with rpk cluster partitions move-status, and see new assignments with rpk topic describe -p TOPIC.

This is an output of of the move command. If this is confusing, we can just remove it from the example. Wdyt? @mattschumpert

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (daisukebe) [reply modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]: Thanks, @ztlpn. Even though 24.1 will have a great improvement on shard assignments it'd still be valuable and worth mentioning that we can specify a shard for a replica, no?

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (daisukebe) [reply modules/manage/pages/cluster-maintenance/cluster-balancing.adoc:87]: Ok, thanks. I will add a note at the bottom of the section.

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (ztlpn) [reply modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]: I'm not sure this functionality has practical value today (were there cases where we or customers had to use it?) and because hopefully it will disappear soon, I'd rather not advertise it today.

Comment (daisukebe) [reply modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]: updated as per our DM, thanks!

File: modules/manage/pages/cluster-maintenance/cluster-balancing.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/cluster-balancing.adoc]:
```suggestion
* Kafka doesn't support shard-level (core) partition assignments, but Redpanda does. For help specifying a shard for partition assignments, see `rpk cluster partitions move --help`.
```

**Comment by Deflaimun** [comment]
@daisukebe if you need to make further adjustments please either

- Track again the branch from remote
- Rebase from remote

This is because we had to rebase this branch, since it was created from 23.2

**Review by micheleRP — APPROVED** [review]
Thank you @daisukebe!

**Pull request merged**
